### DISCUSSION TITLE
Shared client extensions in pnpm workspaces/monorepo projects

### DISCUSSION THREAD

**Discussion by mitchmcleod-ew** [prisma/prisma#19212] [Q&A] [answered]
Is there any guidance on how to use shared client extensions with PNPM monorepos or other project types that contain more than one client?

At the moment we're loving client extensions but can't seem to get around shared extensions necessarily importing the client and ```Prisma``` namespace with which they were made, from ```@prisma/client``` for the ```defineExtension``` call. Even if extensions are written to import from ```.prisma/client``` the resolution pattern tries to resolve to the wrong location (for pnpm everything is relative to the virtual ```node_modules/.pnpm/``` store). 

We've also tried just defining the extension without the ```defineExtension``` call (and then calling define where the client is extended) but then run into max type inference issues as the types are so heavily nested and there are no base types that could be used to build from.

Very useful feature but this is a non-trivial roadblock. Any thoughts or pointers on this would be appreciated.

**Comment by millsp** [comment, marked as answer]
Hey @mitchmcleod-ew, this is spot on because he have not yet provided an official guide wrt. distributing and packaging client extensions. We have an internal issue to create a proper page that explains how to do this. It will be worked on soon probably, but I cannot commit to it. In the mean time, I created a repository here that could help you.

https://github.com/millsp/prisma-local-client-extension-dev

⚠ Please beware that the `@prisma/client/scripts/default-index` is highly subject to change, we will likely change that ⚠

**Reply by nurul3101** [reply]
Hello @mitchmcleod-ew 👋 

Did you had a chance to look at Pierre's response?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
